### PR TITLE
Improve turn instructions when leaving the roads

### LINKED ISSUE #1047: Instructions: Remove Unnecessary Continues
Currently we have unnecessary continues in the instructions, here are some samples:

https://graphhopper.com/maps/?point=48.004367%2C8.137822&point=48.003878%2C8.137844
https://graphhopper.com/maps/?point=48.177999%2C8.572013&point=48.17788%2C8.573037

Currently, we are creating an instruction if we are not sure if an instruction is needed at this point. With the current data it's very complicated to improve the current situation. With highway information we should be able to improve it.

If we are continuing on the same highway class but the outgoing edge is a different highway class, then it is very likely that the turn instruction is not necessary.

Related to #990

### PR DESCRIPTION
**Pull request by boldtrn**
Fixes #1047.

This fixes the two examples given in #1047 and also the new example provided by @karussell.

Let me know what you think about the results? I haven't done too much live testing yet. So feedback would be very welcome.

I also renamed the two edge variable in InstructionsOutgoingEdges again :). I hope the naming is OK? I really struggled with the filteredEdges naming.

### CODE REVIEW

File: core/src/main/java/com/graphhopper/routing/InstructionsFromEdges.java [hunk 1]
```diff
@@ -391,12 +391,12 @@ private int getTurn(EdgeIteratorState edge, int baseNode, int prevNode, int adjN
             forceInstruction = true;
         }
 
-        InstructionsOutgoingEdges outgoingEdges = new InstructionsOutgoingEdges(prevEdge, edge, encoder, maxSpeedEnc, crossingExplorer, nodeAccess, prevNode, baseNode, adjNode);
-        int nrOfPossibleTurns = outgoingEdges.nrOfAllowedOutgoingEdges();
+        InstructionsOutgoingEdges outgoingEdges = new InstructionsOutgoingEdges(prevEdge, edge, encoder, maxSpeedEnc, roadClassEnc, roadClassLinkEnc, crossingExplorer, nodeAccess, prevNode, baseNode, adjNode);
+        int nrOfPossibleTurns = outgoingEdges.nrOfAllowedTurns();
```
Comment (karussell) [inline core/src/main/java/com/graphhopper/routing/InstructionsFromEdges.java]: Can you refactor this to the normal Java style `getAllowedTurns`? (e.g. see GraphHopper.getWorkerThreads)

Comment (boldtrn) [reply core/src/main/java/com/graphhopper/routing/InstructionsFromEdges.java]: Sure, I just pushed this change.

**Comment by boldtrn** [comment]
> Nice! This solves also another problem I recently stumble over when navigating with GraphHopper.

Perfect! Thanks for testing this.

**Comment by karussell** [comment]
Thanks!

**Pull request merged**